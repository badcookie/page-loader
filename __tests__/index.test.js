import { promises as fs } from 'fs';
import path from 'path';
import os from 'os';
import axios from 'axios';
import httpAdapter from 'axios/lib/adapters/http';
import nock from 'nock';
import pageLoader from '../src';

axios.defaults.adapter = httpAdapter;

const filename = 'hexlet-io-courses.html';
const expectedContentFilepath = path.join(__dirname, '__fixtures__', filename);

test('should download resource to appropriate directory', async () => {
  const expectedBody = await fs.readFile(expectedContentFilepath, 'utf-8');

  nock('https://hexlet.io/')
    .get('/courses')
    .reply(200, expectedBody);

  const temporaryDirectory = await fs.mkdtemp(path.join(os.tmpdir(), '/'));
  await pageLoader('https://hexlet.io/courses', temporaryDirectory);
  const actualBody = await fs.readFile(path.join(temporaryDirectory, filename), 'utf-8');
  expect(actualBody).toEqual(expectedBody);
});
