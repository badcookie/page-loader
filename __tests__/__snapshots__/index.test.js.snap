// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should consider error cases 1`] = `"received is not a function"`;

exports[`should consider error cases 2`] = `"received is not a function"`;

exports[`should consider error cases 3`] = `"received is not a function"`;
